lib/src/provider/provider_data.dart

```diff
@@ -1,133 +1,97 @@
-// ignore_for_file: avoid_dynamic_calls, type_annotate_public_apis
-
-import 'package:flutter/cupertino.dart';
-import 'package:flutter/widgets.dart';
-import 'package:supabase_flutter/supabase_flutter.dart';
-import 'package:todo_app/api/api_key.dart';
+import 'package:flutter/material.dart';
 import 'package:todo_app/src/constants.dart';
+import 'package:todo_app/src/models/note.dart';
 
 ///
 class ProviderData with ChangeNotifier {
-  ///
-  // Note? notes;
-  bool loading = false;
-  final int _testd = 0;
+  bool _removeTaskBool = false;
+  Color _textColor = textColorK;
+///
+  List<Note> noteList = [];
+ ///
+  Note? noteInfo;
```

Comment:
Feels like this class should be split into two:
NotesListNotifier
NoteNotifier.

One will manage the full list of tasks, another -- one single note.

Do note that I see this at surface value, I might be wrong.